src/components/Context/webFactory.js

```diff
@@ -12,18 +12,14 @@ governing permissions and limitations under the License.
 
 import { deepAssign } from "../../utils";
 
-export default (window, topFrameSetProvider) => {
-  let topFrameSet;
-
+export default window => {
   return xdm => {
-    topFrameSet = topFrameSet || topFrameSetProvider();
-
     const web = {
       webPageDetails: {
         URL: window.location.href || window.location
       },
       webReferrer: {
-        URL: topFrameSet.document.referrer
+        URL: window.top.document.referrer
```

Comment:
It's interesting to me that we use the top window's referrer instead of the the current window's referrer. I would have assumed that if a user clicked on something inside of an iframe and the iframe loaded up a document I was tracking, I'd want to track the referrer from the iframe rather than the referrer for the window containing the iframe. Do you know why we use the containing window's referrer?

Have you tested this in a cross-origin scenario where the parent window is in a different origin than the iframe's document?

Reply:
The history of the topFrameSet calculation in AppMeasurement.js is lost to time, but reading up on referrers I can guess on why it was done this way. On the initial page load an iframe's referrer will equal the containing page's location (i.e. window.top.location == window.document.referrer).  If we just did `document.referrer` we would lose the outside referrers on a page with frames.  

Integration tests are a good idea to make sure this does the same thing as the old topFrameSet calculation, and see exactly how it behaves with frames. I'll bring it up to @dancingcactus